rust/oak/src/lib.rs

```diff
@@ -96,8 +98,11 @@ pub fn wait_on_channels(handles: &[ReadHandle]) -> Result<Vec<ReadHandle>, OakSt
         }
         let mut results = Vec::with_capacity(handles.len());
         for i in 0..handles.len() {
-            if space[i * wasm::SPACE_BYTES_PER_HANDLE + (wasm::SPACE_BYTES_PER_HANDLE - 1)] != 0 {
-                results.push(handles[i]);
+            let raw_status =
+                space[i * wasm::SPACE_BYTES_PER_HANDLE + (wasm::SPACE_BYTES_PER_HANDLE - 1)];
+            match ChannelReadStatus::from_i32(i32::from(raw_status)) {
```

Comment:
Consider something like

```rust
space.get(i * wasm::SPACE_BYTES_PER_HANDLE + (wasm::SPACE_BYTES_PER_HANDLE - 1))
  .map(i32::from)
  .and_then(ChannelReadStatus::from_i32)
```

Reply:
Done (although it needed `.map(|x| i32::from(*x))` to convert `&u8` to `i32`)

Reply:
That works, or you could use `.cloned()` in between to convert from `&T` to `T`.

Reply:
Done.